---
title: "Between Sample (Beta) Diversity of Microbes along a Salinity Gradient"
author: "Marian Schmidt"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      fig.path = "../figures/05_CommunityAnalysis")
```

# Goals 

1. Load in phyloseq data with rooted tree.  
2. Evaluate sequencing depth and remove sample.  
3. Normalize the read counts between samples.  
4. Calculate community dissimilarities. Numbers between 0 and 1. If 0, completely similar versus if they are 1, then they're completely dissimilar.   
    a. *Sorensen*: Shared Species as a binary value: Abundance-unweighted 
    b. *Bray-Curtis*: Shared Abundant species: Abundance-weighted
    c. *(Abundance-)Weighted UNIFRAC*: Consider Abundant Species and where they fall on the tree  
5. Visualize the community data with two unconstrained Ordinations:  
    a. *PCoA*: Linear Method. Eigenvalue = how much variation is explained by each axis. Choose to view axis 1, 2, 3, etc. and plot them together.  
    b. *NMDS*: Non-linear. Smush multiple Dimensions into 2 or 3 axes. Need to report Stress value (ideally <0.15).  
6. Run statistics with PERMANOVA and betadispR. 


# Setup 

## Load Libraries
```{r load-libraries}
#install.packages("vegan")
pacman::p_load(tidyverse, devtools, phyloseq, patchwork, vegan,
               install = FALSE)

# Load Station colors 
station_colors <- c(
  "Shipping Channel" = "dodgerblue4",
  "Aransas Bay" = "dodgerblue2",
  "Copano West" = "#D9CC3C",
  "Copano East" = "#A0E0BA",
  "Mesquite Bay" = "#00ADA7")
```

## Load Data 
```{r load-physeq}
# Load in rooted phylogenetic tree! 
load("data/03_Phylogenetic_Tree/phytree_preprocessed_physeq.RData")
midroot_physeq_rm456
unrooted_physeq_rm456
```


# Explore Read Counts 
```{r calc-seq-depth}
# Calculate the total number of reads per sample. 
raw_TotalSeqs_df  <- 
  midroot_physeq_rm456 %>%
  # calculate the sample read sums 
  sample_sums() %>%
  data.frame()
# name the column 
colnames(raw_TotalSeqs_df)[1] <- "TotalSeqs"
  
head(raw_TotalSeqs_df)

# make histogram of raw reads 
raw_TotalSeqs_df %>%
  ggplot(aes(x = TotalSeqs)) + 
  geom_histogram(bins = 50) + 
  scale_x_continuous(limits = c(0, 10000)) + 
  labs(title = "Raw Sequencing Depth Distribution") + 
  theme_classic()

```


## Normalize read counts  

# Cacluate & Visualize community dissimilarity 

## Sorensen PCoA

## Bray-Curtis PCoA  

## Weighted Unifrac PCoA 

## Bray-Curtis NMDS 

# Test for Statistical Significance with PERMANOVA & betadispR 

# Session Information 
For reproducibility 
```{r session_info}
devtools::session_info()
```




